pkg/entity/tournament.go

```diff
@@ -0,0 +1,691 @@
+			playerTwo := standings[i+1].Player
+			var newPairing *Pairing
+			// If we are past the first round in an elimination tournament,
+			// the bottom half of the standings have been eliminated.
+			// Each successive round eliminates half as many players,
+			// hence the l / twoPower(l) determines which players are eliminated.
+			if t.PairingMethod == Elimination && round > 0 && i >= l/twoPower(round) {
+				newPairing = newEliminatedPairing(playerOne, playerTwo)
+			} else {
+				newPairing = newClassicPairing(playerOne, playerTwo, t.GamesPerRound)
+			}
+			roundPairings[t.PlayerIndexMap[playerOne]].Pairing = newPairing
+			roundPairings[t.PlayerIndexMap[playerTwo]].Pairing = newPairing
+		}
+	} else if t.PairingMethod == Random {
+		playerIndexes := []int{}
+		for _, v := range t.PlayerIndexMap {
+			playerIndexes = append(playerIndexes, v)
+		}
+		// FIXME: Is this necessary?
+		// It was kind of annoying not to have
+		// nondeterministc results for testing,
+		// but maybe we should have it in production?
+		// rand.Seed(time.Now().UnixNano())
+		rand.Shuffle(len(playerIndexes),
+			func(i, j int) {
+				playerIndexes[i], playerIndexes[j] = playerIndexes[j], playerIndexes[i]
+			})
+		for i := 0; i < len(playerIndexes)-1; i += 2 {
+			newPairing := newClassicPairing(t.Players[playerIndexes[i]], t.Players[playerIndexes[i+1]], t.GamesPerRound)
+			roundPairings[playerIndexes[i]].Pairing = newPairing
+			roundPairings[playerIndexes[i+1]].Pairing = newPairing
+		}
+	} else if t.PairingMethod == RoundRobin {
+		roundRobinPlayers := t.Players
+		// The empty string represents the bye
+		if len(roundRobinPlayers)%2 == 1 {
+			roundRobinPlayers = append(roundRobinPlayers, "")
+		}
+
+		roundRobinPairings := getRoundRobinPairings(roundRobinPlayers, round)
+		for i := 0; i < len(roundRobinPairings)-1; i += 2 {
+			playerOne := roundRobinPairings[i]
+			playerTwo := roundRobinPairings[i+1]
+
+			if playerOne == "" && playerTwo == "" {
+				return errors.New(fmt.Sprintf("Two byes playing each other in round %d\n", round))
+			}
+
+			// The blank string represents a bye in the
+			// getRoundRobinPairings algorithm, but in a Tournament
+			// is represented by a player paired with themselves,
+			// so convert it here.
+
+			if playerOne == "" {
+				playerOne = playerTwo
+			}
+
+			if playerTwo == "" {
+				playerTwo = playerOne
+			}
+
+			newPairing := newClassicPairing(playerOne, playerTwo, t.GamesPerRound)
+			roundPairings[t.PlayerIndexMap[playerOne]].Pairing = newPairing
+			roundPairings[t.PlayerIndexMap[playerTwo]].Pairing = newPairing
+		}
+	}
+	// Give all unpaired players a bye
+	// Byes are always designated as a player
+	// paired with themselves
+	if t.PairingMethod != Manual {
+		for i := 0; i < len(roundPairings); i++ {
+			pri := roundPairings[i]
+			if pri.Pairing == nil {
+				pri.Pairing = newClassicPairing(t.Players[i], t.Players[i], t.GamesPerRound)
+			}
+			if t.PairingMethod == Random {
+
```

Comment:
nothing here?

Reply:
Oops, this is extraneous and is not needed. I will remove it.

Reply:
Done